config/schemas/validationSchemas.js

```diff
@@ -0,0 +1,64 @@
+const z = require("zod");
+
+const usernameSchema = z.string().min(3).max(25);
+
+const basicProfileSchema = z.object({
+  name: z.string().optional(),
+  bio: z.string().optional(),
+  links: z
+    .array(
+      z.object({
+        name: z.string().optional(),
+        url: z.string().url().optional(),
+        username: z.string().optional(),
+        icon: z.string().optional(),
+      })
+    )
+    .optional(),
+});
+
+const socialSchema = z.object({
+  icon: z.string().optional(),
+  url: z.string().url().optional(),
+});
+
+const linkSchema = z.object({
+  group: z.string().optional(),
+  name: z.string().optional(),
+  url: z.string().url().optional(),
+  icon: z.string().optional(),
+  isPinned: z.boolean().optional(),
+});
+
+const testimonialSchema = z.object({
+  title: z.string().optional(),
+  description: z.string().optional(),
+  date: z.string().optional(),
+  username: z.string().optional(),
+  isPinned: z.boolean().optional(),
+});
+
+const milestoneSchema = z.object({
+  isGoal: z.boolean().optional(),
+  title: z.string().optional(),
+  date: z.string().optional(),
```

Comment:
Same here...

```suggestion
  date: z.string().datetime({ offset: true }).optional(),
```

Reply:
I don't think this will work. Everyone has different formats when it comes to the dates. Here are some random examples from the `.json` files for the `milestones`:

```
 "date": "2023",
```

```
"date": "December 2013",
```

```
"date": "Jul 2022",
```

```
"date": "26th September 2022",
```

If I add the validation, many profiles won't be added to the DB. Is that fine?

Reply:
Ok, I did this - https://github.com/EddieHubCommunity/LinkFree/pull/7577#discussion_r1261003242

Reply:
Oh good spot, and great idea 👍